services/src/main/java/org/openlearn/web/rest/PortfolioItemResource.java

```diff
@@ -144,57 +171,110 @@ public ResponseEntity delete(@PathVariable final Long id) {
 	 */
 	@PostMapping(path="/{portfolioId}/upload")
 	@Secured({AuthoritiesConstants.ADMIN, AuthoritiesConstants.ORG_ADMIN, AuthoritiesConstants.INSTRUCTOR, AuthoritiesConstants.STUDENT})
-	public String uploadCourseFile(@PathVariable final Long portfolioId,
+	public ResponseEntity uploadCourseFile(@PathVariable final Long portfolioId,
 								   @RequestParam("file") MultipartFile file,
 								   RedirectAttributes redirectAttributes) throws URISyntaxException {
-		storageService.store(file, null, portfolioId);
-		redirectAttributes.addFlashAttribute("message",
-			"You successfully uploaded " + file.getOriginalFilename() + "!");
-
-		return "You successfully uploaded " + file.getOriginalFilename() + "!";
+		PortfolioItemDTO portfolioItem = portfolioItemService.findOne(portfolioId);
+		if (canUploadFilesToPortfolio(portfolioItem)) {
+			FileInformationDTO response = storageService.store(file, null, portfolioId);
+			redirectAttributes.addFlashAttribute("message",
```

Comment:
What's the point of this flash attribute?

Reply:
Don't know, this isn't actually an addition, it's just a spacing change (because of the higher level `if` I added. I'm not sure why this was originally added or what it does.